calculating_n50_assemblies.py

```diff
@@ -10,69 +10,80 @@ def main(in_dir = None, out_dir = None):
-            
-            #The assembly length is calculated by summing all contig lengths
-            #The length of every contig from the contig length dictionary is written to a file
-            total_assembly_len = 0
-            for key in contig_length_dict:
-                file2.write("\t{}\t{}\n".format(key,contig_length_dict[key]))
-                total_assembly_len += contig_length_dict[key]
-            file2.write("Assembly {} has {} contigs\n\n".format(assembly_name,contig_count))
-            
-            #L50 length is calculated
-            l50_len = total_assembly_len/2
-            
-            #contig length dictionary is sorted by values to an ordered list of tuples
-            rev_sorted_contig_length_dict = sorted(contig_length_dict.items(), key=operator.itemgetter(1),reverse=True)
-            temp_sum = 0
-            
-            #iterate through decreasing contig lengths and breaking the loop once N50 length is calculated
-            for item in rev_sorted_contig_length_dict:
-                temp_sum += item[1]
-                if temp_sum >= l50_len: 
-                    file1.write("N50 length for assembly {} is {}\n".format(assembly_name, item[1]))
-                    break
+                    file2.write("File has assembly named {} with following contigs and respective lengths:\n".format(assembly_name))
+                        
+                    #The assembly length is calculated by summing all contig lengths
+                    #The length of every contig from the contig length dictionary is written to a file
+                    total_assembly_len = 0
+                    for key in contig_length_dict:
+                        file2.write("\t{}\t{}\n".format(key,contig_length_dict[key]))
+                        total_assembly_len += contig_length_dict[key]
+                    file2.write("Assembly {} has {} contigs\n\n".format(assembly_name,contig_count))
+                        
+                    #L50 length is calculated
+                    l50_len = total_assembly_len/2
+                        
+                    #contig length dictionary is sorted by values to an ordered list of tuples
+                    rev_sorted_contig_length_dict = sorted(contig_length_dict.items(), key=operator.itemgetter(1),reverse=True)
+                    temp_sum = 0
+                        
+                    #iterate through decreasing contig lengths and breaking the loop once N50 length is calculated
+                    for item in rev_sorted_contig_length_dict:
+                        temp_sum += item[1]
+                        if temp_sum >= l50_len: 
+                            file1.write("N50 length for assembly {} is {}\n".format(assembly_name, item[1]))
+                            break
+        else:
+            sys.exit("Zipped file contents do not match fasta format extensions https://en.wikipedia.org/wiki/FASTA_format. Exiting...")
```

Comment:
I don't think exiting is the best idea here, it would be better just to print a log statement saying ignoring file etc.